controllers/front/Payment.php

```diff
@@ -479,57 +323,47 @@ private function buildAddresses($request = array())
         $invoicingAddress = new \AddressCore($cart->id_address_invoice);
-        $paymentMethod = \Tools::getValue(self::PAYMENT_METHOD);
 
+        $paymentMethod = \Tools::getValue(self::PAYMENT_METHOD);
+        if (empty($paymentMethod)) {
+            throw new MissingDataException('payment method is not sent in the request!');
+        }
 
-        if (!empty($paymentMethod[self::PERSONAL_DETAILS])) {
-            $personalDetails = $paymentMethod[self::PERSONAL_DETAILS];
-        } else {
-            $personalDetails = array();
+        if (empty($paymentMethod[self::TYPE])) {
+            throw new MissingDataException('payment method type is not sent in the request!');
         }
 
         $paymentMethodType = $paymentMethod[self::TYPE];
         $isOpenInvoice = $this->paymentService->isOpenInvoicePaymentMethod($paymentMethodType);
-
-        // Already Adyen formatted gender code
-        if (!empty($personalDetails[self::GENDER])) {
-            $gender = $personalDetails[self::GENDER];
-        } else {
-            $gender = $this->genderService->getAdyenGenderValueById($customer->id_gender);
-        }
-
         $localeCode = $this->languageAdapter->getLocaleCode($language);
-        if (empty($localeCode)) {
-            $localeCode = '';
-        }
+        $invoicingAddressCountryCode = $this->countryAdapter->getIsoById($invoicingAddress->id_country);
+        $shopperIp = \Tools::getRemoteAddr();
 
-        if (!empty($personalDetails[self::TELEPHONE_NUMBER])) {
-            $telephoneNumber = $personalDetails[self::TELEPHONE_NUMBER];
-        } else {
-            $telephoneNumber = '';
-        }
+        $gender = '';
+        $telephoneNumber = '';
+        $dateOfBirth = '';
 
-        if (!empty($personalDetails[self::DATE_OF_BIRTH])) {
-            $dateOfBirth = $personalDetails[self::DATE_OF_BIRTH];
-        } else {
-            $dateOfBirth = '';
+        if (empty($request[self::PAYMENT_METHOD][self::PERSONAL_DETAILS][self::GENDER])) {
+            $gender = $this->genderService->getAdyenGenderValueById($customer->id_gender);
         }
 
-        $invoicingAddressCountryCode = $this->countryAdapter->getIsoById($invoicingAddress->id_country);
-        if (empty($invoicingAddressCountryCode)) {
-            $invoicingAddressCountryCode = '';
+        if (empty($request[self::PAYMENT_METHOD][self::PERSONAL_DETAILS][self::DATE_OF_BIRTH])) {
+            if ('0000-00-00' !== $customer->birthday) {
+                $dateOfBirth = $customer->birthday;
+            }
         }
 
-        $shopperIp = \Tools::getRemoteAddr();
-        if (empty($shopperIp)) {
-            $shopperIp = '';
+        if (empty($request[self::PAYMENT_METHOD][self::PERSONAL_DETAILS][self::TELEPHONE_NUMBER])) {
+            // Retrieve the phone number from the invoicing address, if phone is null, try phone_mobile
+            $telephoneNumber = $invoicingAddress->phone ? $invoicingAddress->phone:$invoicingAddress->phone_mobile;
```

Comment:
Could both numbers be null?

Reply:
Yes, but then the builder won't take these into account